corehq/apps/style/static/style/ko/components/inline_edit.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Component for an inline editing widget: a piece of text that, when clicked on, turns into a textarea.
+ *
+ * Parameters (all optional)
+ *  - value: Text to display and edit
+ *  - name: HTML name of textarea
+ *  - id: HTML id of textarea
+ *  - placeholder: Text to display when in read-only mode if value is blank
+ *  - inline: Whether or not to display widget in line with surrounding content. Defaults to false.
+ *  - rows: Number of rows in textarea.
+ *
+ * By default, the widget is client-side only, and it is up to the calling code to actually save the value
```

Comment:
sliiiick cant wait to try it out

On Mon, Apr 4, 2016 at 4:47 PM, Daniel Miller <notifications@github.com>
wrote:

> In corehq/apps/style/static/style/ko/components/inline_edit.js
> <https://github.com/dimagi/commcare-hq/pull/11088#discussion_r58445867>:
>
> > @@ -0,0 +1,134 @@
> > +/*
> > + * Component for an inline editing widget: a piece of text that, when clicked on, turns into a textarea.
> > + *
> > + * Parameters (all optional)
> > + *  - value: Text to display and edit
> > + *  - name: HTML name of textarea
> > + *  - id: HTML id of textarea
> > + *  - placeholder: Text to display when in read-only mode if value is blank
> > + *  - inline: Whether or not to display widget in line with surrounding content. Defaults to false.
> > + *  - rows: Number of rows in textarea.
> > + *
> > + * By default, the widget is client-side only, and it is up to the calling code to actually save the value
>
> Guessing this means that by default the user has to click the green
> checkbox after editing and then also submit the form elsewhere on the page.
> This seems like a not great default user experience—I think most users
> would expect the value to be saved after clicking the green checkbox. I'm
> not sure I have any good suggestions to work around that issue, so this is
> mostly just food for thought.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/11088/files/7503fceadd4d6e73fe1aecb57037d3928f95639e#r58445867>
>